### ISSUE TITLE
Calculate Word-Error-Rate, Construct Word-Replacement-Counts LUT

### ISSUE THREAD

**Issue by evamaxfield** [CouncilDataProject/whisper-experiments#5]
Next up on the todo list, given two transcripts (a ground truth and a generated), calculate the following:

* Overall similarity (done in #3)
* Word-Error-Rate (this looks promising: https://github.com/jitsi/jiwer)
* Word-Replacement-Counts LUT (partially done in #3) (example below)
  * general example: `{"original_word": {"replacement_": 3, "replacement_two": 4}}`
  * a look up table of words in the original text and their replacement words in the generated text
  * fake example`{"housing": {"browsing": 3, "rousing": 5}}`
  * the word "housing" in the original text was replaced my "browsing" 3 times and "rousing" 5 times in the generated text.

I personally think we should first try this by joining all transcript sentences together `" ".join(s for s in transcript.sentences)`. I also think we should do this with three variants, the normal text (the joined text snippet from prior sentence), the lowercase version of the normal, and the lowercased and punctuation removed version of the normal.

Basically for every pair of transcripts we send in, we get back something like:

```python
@dataclass
class TranscriptComparison:
    similarity: float
    wer: float  # all three come from that jiwer package I found
    mer: float
    wil: float
    replacements: Dict[str, Dict[str, int]]


@dataclass
class MultiVariantTranscriptComparison:
    raw: TranscriptComparison
    lowercased: TranscriptComparison
    lowercased_no_punct: TranscriptComparison


def compare_transcripts(original: Transcript, generated: Transcript) -> MultiVariantTranscriptComparison:
    pass
```

Thoughts @dphoria ? Can you do take this one?

**Comment by dphoria** [comment]
Ugh I don't know why i didn't see the notification for your mention?! I"m so sorry 🙏 . Would have picked this up way earlier.

**Comment by evamaxfield** [comment]
@dphoria wait. I worked on this a bit. We should chat sometime after the quarter to figure out where to go with this next.

This work is going to turn into a research paper. I am convinced enough that we should just implement whisper into app regardless of this work.

Also I tried some of this and it's... uhhh. hard? i dont really know how to do it hahaha so talking it out together might be good for both of us.
